client/components/jetpack/connection-health/use-check-jetpack-connection-health.ts

```diff
@@ -34,13 +32,21 @@ export const useCheckJetpackConnectionHealth = (
 			persist: false,
 		},
 		staleTime: 10 * 1000,
-		onError: options?.onError,
-		onSuccess: ( data ) => {
-			if ( data?.is_healthy ) {
-				dispatch( setJetpackConnectionHealthy( siteId ) );
-			} else {
-				dispatch( setJetpackConnectionUnhealthy( siteId, data?.error ?? '' ) );
-			}
-		},
 	} );
+
+	const data = query.data;
+
+	useEffect( () => {
+		if ( ! data ) {
+			return;
+		}
+		if ( data.is_healthy && reduxIsUnhealthy ) {
+			dispatch( setJetpackConnectionHealthy( siteId ) );
+		}
+		if ( ! data.is_healthy && ! reduxIsUnhealthy ) {
+			dispatch( setJetpackConnectionUnhealthy( siteId, data.error ?? '' ) );
+		}
+	}, [ dispatch, reduxIsUnhealthy, data, siteId ] );
```

Comment:
Since we only care about `is_healthy` and `error` from `data`, should we assign them to 2 separate variables and use them directly, then add them here as dependencies?

Reply:
Agreed, it's much more comprehensible that way - thanks!